aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -235,10 +229,28 @@ private SyncTime filterOutOldSyncTimes(SyncTime lastSyncTime) throws DataStoreEx
                         processor.onComplete();
                     }
                 })
-                // Flatten the PaginatedResult objects into a stream of ModelWithMetadata objects.
-                .concatMapIterable(PaginatedResult::getItems)
-                // Stop after fetching the maximum configured records to sync.
-                .take(dataStoreConfigurationProvider.getConfiguration().getSyncMaxRecords());
+                // If it's a SerializedModel, add the ModelSchema, since it isn't added during deserialization.
+                .map(paginatedResult -> Flowable.fromIterable(paginatedResult)
```

Comment:
Ah I see, it's a `Flowable` of `Lists` since there can be 0...n pages for a particular model, where each page is a list?

Then on line 145, you flatten each page back into a single Flowable, right?

Reply:
Yep, that's right.  I had intended to batch each "page" of results into one single transaction, which is why this code has been refactored in this way.  I didn't quite get there, in this PR, but plan to do so in the future.